include/aikido/planner/World.hpp

```diff
@@ -0,0 +1,72 @@
+#ifndef AIKIDO_PLANNER_WORLD_HPP_
+#define AIKIDO_PLANNER_WORLD_HPP_
+
+#include <string>
+#include <dart/dart.hpp>
+
+namespace aikido {
+namespace planner {
+
+class World
+{
+public:
+  /// Construct a kinematic World.
+  World(const std::string& name = "");
+
+  virtual ~World();
+
+  /// Create a clone of this World. All Skeletons will be copied over.
+  /// \param newName Name for the cloned World
+  std::shared_ptr<World> clone(const std::string& newName = "") const;
+
+  /// Create a new World inside of a shared_ptr
+  static std::shared_ptr<World> create(const std::string& name = "");
```

Comment:
In case we provide one creator function, then it should be returning `std::unique_ptr` because we can convert `std::unique_ptr` to `std::shared_ptr` but the opposite is not allowed.

Reply:
Sure. I mostly copied this from DART :sweat_smile:

Reply:
Yeah, DART also needs to be tweaked. :disappointed: There is (or was) ongoing discussion: https://github.com/dartsim/dart/pull/845